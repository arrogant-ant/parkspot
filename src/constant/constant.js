export const FORM = {
    USERNAME: 'Username',
    NAME: 'full-name',
    FULLNAME: 'Full Name',
    FIRSTNAME: 'First Name',
    LASTNAME: 'Last Name',
    EMAIL: 'Email',
    CONTACT_NO: 'Contact No.',
};

export const PARKING_FACILITY = {
    SO: {
        BUILDING_ADDR: 'Building Address',
        TOTAL_PARKING: 'Total Parking',
        FACILITIES: 'Facilities',
        FACILITIES_DATA: [
            'Covered',
            'Gated',
            'CCTV Camera',
            'Security Guard',
            '24 Hours Access',
        ],
        LANDMARK: 'Landmark',
    },
    VO: {
        PARKING_TYPE: 'Types of parking',
        PARKING_TYPE_LIST: [
            { id: 0, name: 'Covered Parking' },
            { id: 1, name: 'Open Parking' },
        ],
        DURATION: 'Duration',
        MINIMUM_DURATION_DATA: [
            { id: 0, name: ' < 1 month' },
            { id: 1, name: '2 - 3 months' },
            { id: 2, name: '3 - 5 months' },
            { id: 3, name: 'More than 6 months' },
        ],
        CAR_TYPE: [
            { id: 0, name: 'SUV' },
            { id: 1, name: 'Sedan' },
            { id: 2, name: 'Hatchback' },
        ],
    },
};

export const KYC = {
    OWNER_RADIO_DATA: ['Yes', 'No'],
    DOCUMENT_DATA_SO: [
        { id: 0, name: 'Adhaar Card' },
        { id: 1, name: 'Electricity Bills' },
        { id: 2, name: 'Driving License' },
        { id: 3, name: 'Rent Agreement' },
    ],
    DOCUMENT_DATA_VO: [
        { id: 0, name: 'Adhaar Card' },
        { id: 1, name: 'Driving License' },
    ],
    DOCUMENT_INFO_MSG:
        'Note: Document must contain the address proof of the spot registered.',
};

export const ADD_INFO = {
    AMENITIES: 'Amenities',
    SPOTS: 'Spots',
    MINIMUM_DURATION: 'Minimum Duration',
    RENT: 'Expected Rent',
    AMENITIES_DATA: [
        'Covered',
        'Gated',
        'CCTV Camera',
        'Security Guard',
        'Others',
    ],
    SPOTS_DATA: [
        { id: 0, name: '1' },
        { id: 1, name: '2' },
        { id: 2, name: '3' },
        { id: 3, name: '3+' },
    ],
    MINIMUM_DURATION_DATA: [
        { id: 0, name: 'less than 1 month' },
        { id: 1, name: '2 - 3 months' },
        { id: 2, name: '3 - 5 months' },
        { id: 3, name: 'More than 6 months' },
    ],
    TERMS: 'Terms & Conditions',
    TERMS_DATA: ['I agree to '],
};

export const PREFERENCE = {
    PARKING_TYPE: 'Types of parking',
    DURATION: 'Duration',
    MODEL: 'Car Model',
    PARKING_TYPE_LIST: [
        { id: 0, name: 'Covered Parking' },
        { id: 1, name: 'Open Parking' },
    ],
};

export const ICON = {
    INFO: 'information-outline',
};

export const PAGE_TITLE = {
    TITLE_TEMPLATE: 'ParkSpot | ',
    HOMEPAGE:
        'Find and Book Parking Spaces Nearby | Bangalore Delhi Mumbai Pune Bengaluru | Parkspot.in',
    FAQ: `FAQ's - Get Your All Parking Related Queries be Answered...`,
    ABOUT: 'About -  Get Parking Space , Rent Empty Space',
    CONTACT: 'Contact Us',
    FEATURES: 'Features',
    VO_PORTAL: 'Get Parking Spot',
    SO_PORTAL: 'Register Parking Spot',
    TERMS: 'Terms & Conditions',
    THANK: 'Thank You!',
    ERROR: 'Oops... Something went wrong!!',
    BLOG: 'Blogs',
    SEARCH_PORTAL: 'Search Portal - Admin Only',
    PAYMENT: 'Payment',
    DISCOVER: 'Parking Near ',
    SEARCH: 'ParkSpot | ',
};
