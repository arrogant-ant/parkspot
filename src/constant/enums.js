/**
 *
 * @param {Array} enumLabels
 * @param {int} enumValue
 * @return {string}: enum label for enum value
 */
function getEnumLabel(enumLabels, enumValue) {
    if (enumValue >= enumLabels.length) {
        ('Undefined enum');
    }
    return enumLabels[enumValue];
}

export const PaymentStatus = Object.freeze({
    PaymentNotDefined: 0,
    PaymentPending: 1,
    PaymentIncomplete: 2,
    PaymentFailed: 3,
    PaymentUserDropped: 4,
    PaymentFlagged: 5,
    PaymentSuccess: 6,
    PaymentCancelled: 7,
    PaymentVoid: 8,
    PaymentTransferred: 9,
    PaymentExpired: 10,
    FullRefund: 11,
    PartialRefund: 12,
});

const PaymentStatusLabel = [
    'Not Defined',
    'Pending',
    'Incomplete',
    'Failed',
    'User Dropped',
    'Flagged',
    'Success',
    'Cancelled',
    'Void',
    'Transferred',
    'Expired',
    'Full Refund',
    'Partial Refund',
];

/**
 *
 * @param {int} paymentStatus
 * @return {string}: label for payment Status
 */
export function getPaymentStatusLabel(paymentStatus) {
    return getEnumLabel(PaymentStatusLabel, paymentStatus);
}

export const ParkingRequestStatus = Object.freeze({
    RequestStatusNotSet: 0,
    RequestRegistered: 1,
    RequestProcessing: 2,
    RequestSpotSuggested: 3,
    RequestSpotAccepted: 4,
    RequestSpotDenied: 5,
    RequestArchive: 6,
});

const ParkingRequestStatusLabel = [
    'Not Set',
    'Registered',
    'Processing',
    'Spot Suggested',
    'Spot Accepted',
    'Spot Denied',
    'Archive',
];

/**
 *
 * @param {int} parkingRequestStatus
 * @return {string}: label for parking request status
 */
export function getParkingRequestStatus(parkingRequestStatus) {
    return getEnumLabel(ParkingRequestStatusLabel, parkingRequestStatus);
}

export const KYCStatus = Object.freeze({
    // NotSet => KYC information NOT available
    NotSet: 0,
    // Pending => pending for document verification
    Pending: 1,
    // IDVerified => User ID card is successfully verified
    IDVerified: 2,
    // OwnershipVerified => User vehicle/spot ownership (depending on UserType) is successfully verified
    OwnershipVerified: 3,
    // Verified => all documents successfully KYC verified
    Verified: 4,
    // Denied => process started but non of the document was verified
    Denied: 5,
    // NotVerified => KYC process has not even started
    NotVerified: 6,
});

const KYCStatusLabel = [
    'NotSet',
    'Pending',
    'IDVerified',
    'OwnershipVerified',
    'Verified',
    'Denied',
    'NotVerified',
];

/**
 *
 * @param {int} kyStatus
 * @return {string}: label for kyc status
 */
export function getKYCStatusLabel(kyStatus) {
    return getEnumLabel(KYCStatusLabel, kyStatus);
}

export const BookingStatus = Object.freeze({
    // BookingInitiated => fired tentative, SO not confirmed yet.
    BookingInitiated: 0, // tentative
    BookingConfirmed: 1, // active
    BookingCancelled: 2, // past
    BookingRefunded: 3, // past
    // BookingFailedUnavailiblity => spot NOT available or owner declined.
    BookingFailedUnavailiblity: 4, // tentative
    BookingFailedError: 5,
    // BookingPaymentPending => SO confirmed waiting for payment.
    // only incase when visit is not needed.
    BookingPaymentPending: 6, // active
    // BookingVisiting => VO has paid conv fee. VO will visit/visited the spot.
    BookingVisiting: 7,
    // SO confirmed waiting for payment to schedule visit
    BookingScheduleVisit: 8,
    // Booking in progress but upcoming rent is due.
    BookingRentDue: 9,
});

export const BookingStatusLabels = [
    'Initiated',
    'Confirmed',
    'Cancelled',
    'Refunded',
    'Unavailable',
    'Failed Error',
    'Payment Pending (Conv Fee + Rent + SD)',
    'Visiting (Rent + SD)',
    'Schedule Visit (Conv fee due)',
    'Rent Due',
];

/**
 *
 * @param {int} bookingStatus
 * @return {string}: label for booking status
 */
export function getBookingStatusLabel(bookingStatus) {
    return getEnumLabel(BookingStatusLabels, bookingStatus);
}

export const PaymentPeriodicity = Object.freeze({
    PeriodicityOnce: 0,
    PeriodicityDaily: 1,
    PeriodicityMonthly: 2,
});

export const PaymentPeriodicityLabels = ['Once', 'Daily', 'Monthly'];

/**
 *
 * @param {int} paymentPeriodicity
 * @return {string}: label for payment periodicity
 */
export function getPaymentPeriodicityLabel(paymentPeriodicity) {
    return getEnumLabel(PaymentPeriodicityLabels, paymentPeriodicity);
}

export const PaymentType = Object.freeze({
    PaymentTypeNotSet: 0,
    ConvenienceFee: 1,
    MonthlyRentWithConvenienceFeeAndSecurityDeposit: 2,
    MonthlyRentAndSecurityDeposit: 3,
    MonthlyRent: 4,
});

export const PaymentTypeLabels = [
    'PaymentTypeNotSet',
    'ConvenienceFee',
    'MonthlyRentWithConvenienceFeeAndSecurityDeposit',
    'MonthlyRentAndSecurityDeposit',
    'MonthlyRent',
];

/**
 *
 * @param {int} paymentType
 * @return {string}: label for payment Type
 */
export function getPaymentTypeLabel(paymentType) {
    return getEnumLabel(PaymentTypeLabels, paymentType);
}
