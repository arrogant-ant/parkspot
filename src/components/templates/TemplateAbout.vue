<template>
    <BodyWrapper>
        <div class="about-container">
            <AtomImage
                :src="aboutImg"
                :alt="'About Image'"
                class="about-img"
            ></AtomImage>
            <div class="about-desc" data-aos="zoom-in">
                <AtomHeading class="mb-3 ml-1"> Why Choose Us? </AtomHeading>
                <AtomParagraph class="mb-2 ml-1 mr-1">
                    Comfortable parking experience. Brokerage free. Genuine
                    Owners. Ensured security of your vehicle. Available for
                    residents and visitors. Surveillance parking. Simple steps
                    to list property. Earn money on the go. Available on app and
                    website.
                </AtomParagraph>
                <AtomParagraph class="mb-6 ml-1 mr-1">
                    ParkSpot is part of this new initiative and incorporates
                    various features aimed at improving the parking experience
                    for residents and visitors. These will include removing the
                    need for entry barriers at car park entrances and exits for
                    a seamless flow of traffic, as vehicle plates will be
                    registered via cameras.
                </AtomParagraph>
                <div class="about-links-container">
                    <div class="about-links">
                        <p class="park-link">Get ParkSpot App</p>
                        <div class="app-links">
                            <span>
                                <a
                                    :href="APP_LINK.ANDROID"
                                    target="_blank"
                                >
                                    <figure class="card-img image">
                                        <img
                                            alt="google app link"
                                            :src="appLinkImg"
                                        />
                                    </figure>
                                </a>
                            </span>
                            <span>
                                <a :href="APP_LINK.IOS" target="_blank">
                                    <figure class="card-img image">
                                        <img
                                            alt="ios app link"
                                            :src="iosAppLinkImg"
                                        />
                                    </figure>
                                </a>
                            </span>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    </BodyWrapper>
</template>

<script>
import BodyWrapper from '../extras/BodyWrapper.vue';
import AtomHeading from '../atoms/AtomHeading.vue';
import AtomParagraph from '../atoms/AtomParagraph.vue';
import AtomImage from '../atoms/AtomImage.vue';
import { APP_LINK } from '@/constant/constant';
export default {
    name: 'TemplateAbout',
    components: {
        BodyWrapper,
        AtomHeading,
        AtomParagraph,
        AtomImage,
    },
    data() {
        return {
            appLinkImg: require('@/assets/google-play-badge.svg'),
            iosAppLinkImg: require('@/assets/apple-store.svg'),
            aboutImg: require('@/assets/about.svg'),
            APP_LINK,
        };
    },
};
</script>

<style lang="scss" scoped>
.about-links {
    display: flex;
    align-items: center;
    gap: 50px;

    @media (max-width: 1024px) {
        gap: 5px;
    }
    .app-links {
        display: flex;
        gap: 8px;
        width: 300px; 
    }
}

.about-links .park-link {
    font-size: 1.5rem;
}

@media only screen and (max-width: 1024px) {
    .about-links .park-link {
        font-size: 1rem;
    }
}

.about-container {
    display: flex;
    align-items: center;
    gap: 2rem;
}

@media only screen and (max-width: 750px) {
    .about-container {
        flex-direction: column;
    }
}

.about-img {
    flex: 30%;
}

.about-desc {
    flex: 50%;
}
</style>
