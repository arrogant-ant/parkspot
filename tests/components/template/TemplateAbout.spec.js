import { mount } from '@vue/test-utils'
import { describe, it, expect } from 'vitest'
import TemplateAbout from '@/components/templates/TemplateAbout.vue'
import { APP_LINK } from '@/constant/constant'

const stubs = {
  BodyWrapper: { template: '<div><slot /></div>' },
  AtomHeading: { template: '<h3><slot /></h3>' },
  AtomParagraph: { template: '<p><slot /></p>' },
  AtomImage: { template: '<img />' },
}

describe('TemplateAbout.vue (user-focused + style checks)', () => {
  let wrapper

  beforeEach(() => {
    wrapper = mount(TemplateAbout, { global: { stubs } })
  })

  it('renders the company overview text', () => {
    expect(wrapper.text()).toContain('Company Overview:')
    expect(wrapper.text()).toContain('Welcome to ParkSpot')
  })

  it('renders the services section', () => {
    expect(wrapper.text()).toContain('Our Services:')
    expect(wrapper.text()).toContain('monthly car parking services')
  })

  it('renders navigation links for parking spot', () => {
    const links = wrapper.findAll('a')
    const hrefs = links.map(a => a.attributes('href'))
    expect(hrefs).toContain('/get-parking-spot')
    expect(hrefs).toContain('/register-parking-spot')
  })

  it('has app store links for Android and iOS', () => {
    const links = wrapper.findAll('.app-links a')
    expect(links.length).toBe(2)
    expect(links[0].attributes('href')).toBe(APP_LINK.ANDROID)
    expect(links[1].attributes('href')).toBe(APP_LINK.IOS)
  })

  it('renders app store images with correct alt text', () => {
    const imgs = wrapper.findAll('.app-links img')
    expect(imgs.length).toBe(2)
    expect(imgs[0].attributes('alt')).toBe('google app link')
    expect(imgs[1].attributes('alt')).toBe('ios app link')
  })

  it('renders the main about image (existence check)', () => {
    const img = wrapper.find('.about-img')
    expect(img.exists()).toBe(true)
  })

  // CSS class checks
  it('applies correct classes for layout and links', () => {
    const container = wrapper.find('.about-container')
    expect(container.classes()).toContain('about-container')

    const desc = wrapper.find('.about-desc')
    expect(desc.classes()).toContain('about-desc')

    const links = wrapper.find('.about-links')
    expect(links.classes()).toContain('about-links')

    const appLinks = wrapper.find('.app-links')
    expect(appLinks.classes()).toContain('app-links')

    const parkLink = wrapper.find('.park-link')
    expect(parkLink.classes()).toContain('park-link')
  })

  // Snapshot test
  it('matches the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})