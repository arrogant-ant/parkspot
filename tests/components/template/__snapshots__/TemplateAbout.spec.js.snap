// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TemplateAbout.vue (user-focused + style checks) > matches the snapshot 1`] = `
"<div data-v-0a89d346="">
  <div data-v-0a89d346="" class="about-container"><img data-v-0a89d346="" src="/assets/about.svg" alt="About Image" class="about-img">
    <div data-v-0a89d346="" class="about-desc" data-aos="zoom-in">
      <h3 data-v-0a89d346="" class="mb-3 ml-1" level="h3">Company Overview:</h3>
      <p data-v-0a89d346="" class="mb-2 ml-1 mr-1"> Welcome to ParkSpot, your ultimate destination for hassle-free monthly car parking solutions. At ParkSpot, our mission is to revolutionize the parking experience by providing convenient and reliable monthly car parking services. We value efficiency, reliability, and customer satisfaction above all else. </p>
      <hr data-v-0a89d346="">
      <h3 data-v-0a89d346="" class="mb-3 ml-1" level="h3">Our Services:</h3>
      <p data-v-0a89d346="" class="mb-2 ml-1 mr-1"> ParkSpot offers a range of monthly car parking services designed to meet your needs. With our user-friendly platform, finding and booking a parking spot is quick and easy. Experience the convenience and benefits of secure parking tailored to your requirements. </p><br data-v-0a89d346="">
      <p data-v-0a89d346="" class="mb-2 ml-1 mr-1"> Let's get started and <a data-v-0a89d346="" href="/get-parking-spot">get your Parking Spot</a> or <a data-v-0a89d346="" href="/register-parking-spot">register your Parking Spot</a> with your parking companion. </p>
      <div data-v-0a89d346="" class="about-links-container">
        <div data-v-0a89d346="" class="about-links">
          <p data-v-0a89d346="" class="park-link">Get ParkSpot App</p>
          <div data-v-0a89d346="" class="app-links"><span data-v-0a89d346=""><a data-v-0a89d346="" href="https://play.google.com/store/apps/details?id=com.parkspot.maya_nagri&amp;utm_source=parkspot.in&amp;utm_campaign=website" target="_blank"><figure data-v-0a89d346="" class="card-img image"><img data-v-0a89d346="" alt="google app link" src="/assets/google-play-badge.svg"></figure></a></span><span data-v-0a89d346=""><a data-v-0a89d346="" href="https://apps.apple.com/in/app/parkspot-find-book-parking/id6449634064" target="_blank"><figure data-v-0a89d346="" class="card-img image"><img data-v-0a89d346="" alt="ios app link" src="/assets/apple-store.svg"></figure></a></span></div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
